### DISCUSSION TITLE
could not find 'schema' in the crate root

### DISCUSSION THREAD

**Discussion by dogou77** [diesel-rs/diesel#3081] [Q&A] [answered]
Hello, I have been following a tutorial on how to build a REST API using Diesel and Rust (https://betterprogramming.pub/rest-api-in-rust-step-by-step-guide-b8a6c5fcbff0)

However, after I finished setting up the migrations (I used a different table than from the example, I am using a table called `logs` instead of `users`) and started working on my models.rs file, when I implemented `use super::schema::logs;` it gave me an error.

I have several pictures attached showing the error output, models.rs, and schema.rs

![Screenshot (49)](https://user-images.githubusercontent.com/9262186/157959020-45f21575-8977-49d9-bb75-78fbe8b3821c.png)

![Screenshot (46)](https://user-images.githubusercontent.com/9262186/157958704-50a189a9-5b80-4191-a862-fc3519dac6e1.png)

![Screenshot (47)](https://user-images.githubusercontent.com/9262186/157958730-d2c68d32-ff65-4bde-b6c4-73ab90caa368.png)

Any help would be appreciated. Thank you.

**Comment by weiznich** [comment, marked as answer]
Your question is missing relevant details. Without an reproducible example its really hard to answer such questions. Additionally, pictures are a really bad way to post code or error messages, as their content cannot be searched or copied.
That written: The error message indicates that rustc could not find the schema module in the specified location. So maybe your code misses a `mod schema` or your schema module is just located somewhere else. Maybe checkout the rust book chapter about the module system again?

**Reply by dogou77** [reply]
Terribly sorry, I am new to GitHub discussions. What relevant details do you need? I am happy to help provide you more details.

**Reply by weiznich** [reply]
> What relevant details do you need?

To answer questions about this kind of error I do need information about your module structure. The easiest way  to provide those, is to provide a complete example project, where someone could just use `cargo check` to reproduce your problem.

**Reply by dogou77** [reply]
Upon further examination of my code, I noticed that I had for whatever reason commented out my `mod schema` in `main.rs` like you said. Thanks for your help, it works just fine now.
